### PR TITLE
Dell Optiplex 7019/9010 SFF (and DT form factor) + TXT inclusion based on coreboot master 24.02.01 for now

### LINKED ISSUE #1739: Port Optiplex 7010/9010 based on Dasharo coreboot fork
Track
- https://github.com/Dasharo/coreboot/pull/541
- https://github.com/Dasharo/dasharo-issues/milestone/6
- https://matrix.to/#/!pAlHOfxQNPXOgFGTmo:matrix.org/$D3k3qBFgILsFCmTwz77aZN0WaBUDlJpU4YcB00_kAzQ?via=matrix.org&via=nitro.chat&via=envs.net

This board could get Heads basic maintenance covererage through  "Dasharo Pro package" once board supported under Heads. 

Already bought this platform in December 2023 but didn't gave it any love up to now.

### PR DESCRIPTION
**Pull request by tlaurion**
This pull request includes various updates and additions to support the Dell Optiplex 7010/9010 systems, including new workflows, configuration files, and scripts for handling firmware blobs. The most important changes include updating the Docker image version, adding new jobs and workflows in the CircleCI configuration, and introducing new configuration files and scripts for the Optiplex 7010/9010 systems.

### CircleCI Configuration Updates:
* Updated Docker image version to `tlaurion/heads-dev-env:v0.2.3` in multiple jobs (`prep_env`, `build_and_persist`, `build`, `save_cache`). [[1]](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L48-R48) [[2]](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L114-R118) [[3]](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L142-R146) [[4]](diffhunk://#diff-78a8a19706dbd2a4425dd72bdab0502ed7a2cef16365ab7030a5a0588927bf47L163-R167)
* Added a new job to download Optiplex 7010/9010 blobs.
* Introduced new workflows for building various configurations of the Optiplex 7010/9010.

### New Configuration Files:
* Added configuration files for different Optiplex 7010/9010 builds (`optiplex-7010_9010-hotp-maximized.config`, `optiplex-7010_9010-maximized.config`, `optiplex-7010_9010_TXT-hotp-maximized.config`, `optiplex-7010_9010_TXT-maximized.config`). [[1]](diffhunk://#diff-bdda7ef01145d01ecf1999aafab35d5a6a286f7cf30cda1ee545086bd744aa8fR1-R93) [[2]](diffhunk://#diff-62051791f14a4d3a4327d3f022f9fd21bb3aedcb398cd17d1e51dda809a47e04R1-R93) [[3]](diffhunk://#diff-67a2e2cb6adb2861636bd135ccdbc59f53e51d0cd9b862d325c454375198bd7fR1-R93) [[4]](diffhunk://#diff-2406de558247565a482b2c9efe9fc44ea924d13381f78a9ed62a2810b9236216R1-R93)

### New Scripts and Documentation:
* Added a script to download and verify required blobs for Optiplex 7010/9010 (`optiplex_7010_9010.sh`).
* Added README and layout files for Optiplex 9010 blobs. (`README.md`, `layout.txt`). [[1]](diffhunk://#diff-e3bfda88fa162d1c2bc55d41f55466e6439477713253e249a286efff354970f8R1-R16) [[2]](diffhunk://#diff-8206d1abe34d8b557ab2b8b90b2d0023cd56e014febeb931827e8867e7a5dac8R1-R4)

### Other Changes:
* Updated `.gitignore` to include new blob files.
* Updated `flake.nix` to include new dependencies and ensure root entries in `/etc/passwd` and `/etc/group`. [[1]](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R90-R91) [[2]](diffhunk://#diff-206b9ce276ab5971a2489d75eb1b12999d4bf3843b7988cbe8d687cfde61dea0R145-R150)
* Added makefile targets for downloading Optiplex 7010/9010 blobs. (`optiplex_blobs.mk`).

----

This is PoC state. Please comment.

- Download extracts and use automatically:
  - IFD maximized size from xx30 blobs extraction script
  - ME neutered with minimal size per xx30 blobs extraction script
  - **TXT variant* only*: ACM (IVB_BIOSAC_PRODUCTION.bin) + SINIT (SNB_IVB_SINIT_20190708_PW.bin)
    - TXT versioning taken from https://github.com/Dasharo/coreboot/blob/7226ebce6e8479b9aef6d4e4ea2420fffa5f75a6/configs/config.dell_optiplex_9010_sff_txt#L10C1-L10C31 
  - EC firmware (sch5545_ecfw.bin)
  - Needs to be flashed through recovery method as all maximized boards

Let's see if that works. Optiplex board owners out here?

----

At completion, will fix #1739, possibly moving to dasharo fork (doesn't build with novacusstom nv41 fork today) to be evaluated (v0.2.0 older then 0.1.0? Confused by versioning of fork: was told to use coreboot master since no change and should work)

 
---

Next steps
- ~Create blobs extraction scripts automating https://docs.dasharo.com/variants/dell_optiplex/initial-deployment/#firmware-preparation~: done
- ~Iterate to enabling TXT with another download/extraction blobs (ACM/sINIT)~: done
- [ ] Testing for what is expected to work by Dasharo team @macpijan @pietrushnic
  - [x] test https://docs.dasharo.com/variants/dell_optiplex/recovery/
    - [x] ~**_Unfortunately I do not own a SOIC-16 clip_** So I cannot recover from a brick~ : EDIT I have DT variant, 2xsoic8
  - [x] Iterate until working as expected
  - [x] Put PR back to ready for review
  - [x] Have review+approval from at least one of @macpijan @pietrushnic @miczyg1 
  - [ ] Merge
  - [ ] Update latest docker image -> v0.2.3

### CODE REVIEW

**Comment by tlaurion** [comment]
Duh.  f2e0553 doesn't contain new flake.nix requirements for blobs extraction. Pushing new docker image in new commit using it.

**Comment by tlaurion** [comment]
boots, but symbols missing. kernel works but most everything depending on initrd seems corrupt. Seems like I missed cleaning up ifd to make its region match expected ones or something, relaunching build without cache to compare roms produced. 

Past CI roms createsd soft brick.

Some notes: my 9010 had service jumper slot easily availabe without needing to disassemble anything other then opening the chassis. DTS then was able to take a backup and flash through internal flashing of flashrom through DTS booted from USB.

Putting in draft.

**Comment by tlaurion** [comment]
Damn. IFD extracted from board's ME region doesn't match ME region from backup :/ More work needed

Log https://app.circleci.com/pipelines/github/tlaurion/heads/2880/workflows/ff9dee9b-9e99-4514-afa6-db3a82e60b08/jobs/54332

**Comment by tlaurion** [comment]
Notes: my 9010 board is way different then 7010 board pictures under Dasharo docs

- Service jumper easily accessible (red)
- Soic8 SPI flash chips also easily accessible (green)

Pictures:
signal-2024-10-08-141307
signal-2024-10-08-140906

**Comment by tlaurion** [comment]
Installing QubesOS on the Optiplex, externally flashed local build rom, booted from detached signed ISO: ok.

**Comment by tlaurion** [comment]
- HOTP variants seal HOTP secret on usb security dongle and enforces automatic boot of default entry
- TPM DUK unseals secret and boots into QubesOS

**Comment by tlaurion** [comment]
@miczyg1 please test internal firmware upgrade and approve this PR

**Comment by miczyg1** [comment]
Okay, will try to do this ASAP.

**Comment by miczyg1** [comment]
> @miczyg1 please test internal firmware upgrade and approve this PR upon successful testing

Heads firmware update using the ZIP from Circle CI (commit https://github.com/linuxboot/heads/pull/1785/commits/d3d4247903fa0d0fb765d440b16308040c9f9db8) worked like a charm.

File: config/coreboot-optiplex-7019_9010-maximized.config [hunk 1]
```diff
@@ -0,0 +1,687 @@
...
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_8 is not set
+CONFIG_DEFAULT_CONSOLE_LOGLEVEL_7=y
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_6 is not set
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_5 is not set
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_4 is not set
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_3 is not set
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_2 is not set
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1 is not set
+# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_0 is not set
+CONFIG_DEFAULT_CONSOLE_LOGLEVEL=7
```
[the remark this answers is not in the document]
Comment (miczyg1) [reply config/coreboot-optiplex-7019_9010-maximized.config:579]:
```suggestion
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_7 is not set
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_6 is not set
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_5 is not set
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_4 is not set
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_3 is not set
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_2 is not set
# CONFIG_DEFAULT_CONSOLE_LOGLEVEL_1 is not set
CONFIG_DEFAULT_CONSOLE_LOGLEVEL_0=y
CONFIG_DEFAULT_CONSOLE_LOGLEVEL=0
```

**Review by miczyg1 — APPROVED** [review]

**Comment by tlaurion** [comment]
Well, archive.org has hiccups after being hacked...

File: config/coreboot-optiplex-7019_9010-maximized.config [hunk 2]
```diff
@@ -568,23 +568,23 @@ CONFIG_CONSOLE_CBMEM=y
...
+CONFIG_DEFAULT_CONSOLE_LOGLEVEL=0
 CONFIG_CONSOLE_USE_LOGLEVEL_PREFIX=y
 CONFIG_CONSOLE_USE_ANSI_ESCAPES=y
 # CONFIG_CMOS_POST is not set
 CONFIG_POST_DEVICE_NONE=y
 # CONFIG_POST_DEVICE_LPC is not set
 # CONFIG_POST_DEVICE_PCI_PCIE is not set
 CONFIG_POST_IO_PORT=0x80
-CONFIG_HWBASE_DEBUG_CB=y
+CONFIG_HWBASE_DEBUG_NULL=y
```
Comment (tlaurion) [inline config/coreboot-optiplex-7019_9010-maximized.config:587]: @miczyg1 also turns this on saving in oldconfig

**Comment by miczyg1** [comment]
@tlaurion I am unable to resolve conversations on my own here. Please feel free to resolve the comments

File: config/coreboot-optiplex-7019_9010-maximized.config (hunk 2 above)
Comment (tlaurion) [reply config/coreboot-optiplex-7019_9010-maximized.config]: @miczyg1 Actually, if serial console output is the thing that slows things down (as discussed under #1806) then only serial console output should be deactivated. https://github.com/linuxboot/heads/compare/c5bbbe38fa8f6d4e5e92f7199dda7191245815a7..2424067203617b23f10ed48b65b8dea037e7b58b seems fine?

File: config/coreboot-optiplex-7019_9010-maximized.config (hunk 1 above)
[the remark this answers is not in the document]
Comment (tlaurion) [reply config/coreboot-optiplex-7019_9010-maximized.config:560]: @miczyg1 https://github.com/linuxboot/heads/compare/c5bbbe38fa8f6d4e5e92f7199dda7191245815a7..2424067203617b23f10ed48b65b8dea037e7b58b preferred?

**Comment by tlaurion** [comment]
> @tlaurion I am unable to resolve conversations on my own here. Please feel free to resolve the comments

Seems like https://github.com/linuxboot/heads/compare/c5bbbe38fa8f6d4e5e92f7199dda7191245815a7..2424067203617b23f10ed48b65b8dea037e7b58b  should be preferred. Commented in previous discussions. Please thumb up and I will resolve comment threads.

File: config/coreboot-optiplex-7019_9010-maximized.config (hunk 2 above)
Comment (miczyg1) [reply config/coreboot-optiplex-7019_9010-maximized.config]: Disabling serial console is fine as well.

**Pull request merged**